bf2fc6cc711aee1a0c2a/kas-fleet-manager
# Preliminary draft work for Processors

## Description

This is some preliminary work to add the basics for `Processors` to `cos-fleet-manager`.

It is a work in progress to get some feedback.

It lacks:
- Implementation of the _worker_ mechanism for `Processor'`s.
- Tests

## Verification Steps

1. Fetch this PR
2. Fetch `cos-fleet-manager`
3. Patch `cos-fleet-manager` to use this version of `kas-fleet-manager` by adding this to its `go.mod` file:
`replace github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager v0.0.0-20230120171214-d0c418df2eee => ../kas-fleet-manager`
4. Run `cos-fleet-manager`
5. Make a Cluster
6. Make a Namespace
7. Run `make run/docs` in `kas-fleet-manager`
8. See the new `Processor` endpoints. `Get`, `List`, `Create` and `Delete` are supported up to and including the database.

## Checklist (Definition of Done)
